### PR TITLE
fix: use `getNestedConstraints()` instead of `getNestedContraints` when available

### PR DESCRIPTION
**Pull request by azjezz**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaCollectionRestriction.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ public function supports(Constraint $constraint, ApiProperty $propertyMetadata):
     private function mergeConstraintRestrictions(Constraint $constraint, ApiProperty $propertyMetadata): array
     {
         $propertyRestrictions = [];
-        $nestedConstraints = method_exists($constraint, 'getNestedContraints') ? $constraint->getNestedContraints() : $constraint->constraints;
+        $nestedConstraints = method_exists($constraint, 'getNestedContraints') ? $constraint->getNestedContraints() : $constraint->getNestedConstraints();
```
Comment (norkunas) [inline src/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaCollectionRestriction.php:84]: Isn't it the same now?

Comment (soyuka) [reply src/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaCollectionRestriction.php]: the method exists is there for backward compatibility

**Comment by mbrodala** [comment]
See https://github.com/symfony/validator/commit/372c74ac5d7b41bd2f313d6f069131b22acd569c

File: src/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaOneOfRestriction.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ public function __construct(iterable $restrictionsMetadata = [])
      */
     public function create(Constraint $constraint, ApiProperty $propertyMetadata): array
     {
-        $oneOfConstraints = $constraint->getNestedContraints();
+        $oneOfConstraints = method_exists($constraint, 'getNestedContraints') ? $constraint->getNestedContraints() : $constraint->constraints;
```
Comment (mbrodala) [inline src/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaOneOfRestriction.php:45]:
```suggestion
        $oneOfConstraints = method_exists($constraint, 'getNestedContraints') ? $constraint->getNestedContraints() : $constraint->getNestedConstraints();
```

Why not like this?

Comment (azjezz) [reply src/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaOneOfRestriction.php:45]: some constraints don't seem to have this method, not really sure *how*, but tests fail when using the nested constraints method.

**Comment by soyuka** [comment]
thanks!

**Pull request merged**